### PR TITLE
DeployApplicationVersion Role based Access Control 

### PR DESCRIPTION
**Pull request by bernardo-bastos**
Implements RBAC on DeployApplicationVersion. Only activated if DEX is enabled.

### CODE REVIEW

File: services/cd-service/pkg/repository/transformer.go [hunk 1]
```diff
@@ -844,13 +844,18 @@ func (c *QueueApplicationVersion) Transform(ctx context.Context, state *State) (
...
 type DeployApplicationVersion struct {
+	Authentication
 	Environment   string
 	Application   string
 	Version       uint64
 	LockBehaviour api.LockBehavior
 }
 
 func (c *DeployApplicationVersion) Transform(ctx context.Context, state *State) (string, error) {
+	err := state.checkUserPermissions(ctx, c.Environment, "*", auth.PermissionDeployRelease, c.RBACConfig)
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer.go:855]:
```suggestion
	err := state.checkUserPermissions(ctx, c.Environment, c.Application, auth.PermissionDeployRelease, c.RBACConfig)
```

Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/transformer.go:855]: I think we need the app here. Not everyone is allowed to deploy all apps.

**Review by sven-urbanski-freiheit-com — CHANGES_REQUESTED** [review]
.

File: services/cd-service/pkg/repository/transformer.go (hunk 1 above)
Comment (bernardo-bastos) [reply services/cd-service/pkg/repository/transformer.go]: Yes correct, I added the app check here.

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**